xbmc/utils/StringUtils.cpp

```diff
@@ -1295,7 +1295,7 @@ std::string StringUtils::FormatFileSize(uint64_t bytes)
 
   unsigned int i = 0;
   double value = static_cast<double>(bytes);
-  while (i < static_cast<int>(units.size()) - 1 && value >= 999.5)
+  while (i < units.size() - 1 && value >= 999.5)
```

Comment:
i should be std::size_t and not unsigned int otherwise it'll generate warnings as size_t can be 32/64 bit